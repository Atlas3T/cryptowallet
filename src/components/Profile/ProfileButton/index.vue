<template>
  <q-page>
    <div class="profileButton">
      <div class="container">
        <router-link
          id="btn_id"
          tag="button"
          to="/UserProfile">
          <p id="profile_id">{{ $t('profile') }}</p>
        </router-link>
      </div>
    </div>
  </q-page>
</template>

<script>

export default {

  // data() {
  //   return {

  //   };
  // },

  // beforeCreate() {

  // },

  // created() {

  // },

  // beforeMount() {

  // },

  // mounted() {

  // },

  // beforeUpdate() {

  // },

  // updated() {

  // },

  // beforeDestroy() {

  // },

  // destroyed() {

  // },

  // methods: {

  // },

};

</script>

<style scoped>
#profile_id{
  position:relative;
  left: 0%;
  top: 35%;
  font-weight:bold;
}

.container {
    position: relative;
}

#btn_id {
    background-image: url( "~assets/avatar.png" );
    background-color: Transparent;
    background-size: 90px 95px;
    height: 110px;
    width: 95px;
    background-repeat:no-repeat;
    border: none;
    cursor:pointer;
    overflow: hidden;
    outline:none;
    }
</style>
