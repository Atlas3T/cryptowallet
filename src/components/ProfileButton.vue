<template>
  <q-page>
    <div class="profileButton">
      <div class="container">
        <router-link
          id="Btn_id"
          tag="button"
          to="/UserProfile">
          <p id="profile_id">Profile</p>
        </router-link>
      </div>
    </div>
  </q-page>
</template>

<script>

export default {

  data() {
    return {

    };
  },

  beforeCreate() {

  },

  created() {

  },

  beforeMount() {

  },

  mounted() {

  },

  beforeUpdate() {

  },

  updated() {

  },

  beforeDestroy() {

  },

  destroyed() {

  },

  methods: {

  },

};

</script>

<style @scoped>
#profile_id{
  position:relative;
  left: 0%;
  top: 35%;
  font-weight:bold;
}

.container {
    position: relative;
    /* width: 10%; */
    /* max-width: 500px; */
}

#Btn_id {
    background-image: url( "~assets/avatar.png" );
    background-color: Transparent;
    background-size: 90px 95px;
    height: 110px;
    width: 95px;
    background-repeat:no-repeat;
    border: none;
    cursor:pointer;
    overflow: hidden;
    outline:none;
    }
</style>
