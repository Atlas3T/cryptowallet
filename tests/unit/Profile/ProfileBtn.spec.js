import { shallowMount } from '@vue/test-utils';
import ProfileButton from '@/components/Profile/ProfileButton.vue';
import { localVue } from '../setupLocalVue';
import VueI18n from 'vue-i18n'

describe('ProfileButton.vue', () => {
  let wrapper;

  function wrapperInit(options) {
    return shallowMount(ProfileButton, options);
  }

  beforeEach(() => {
    const i18n = new VueI18n({locale: 'en-us'});
    wrapper = wrapperInit({ localVue, i18n });
  });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });


  it('expect main content', () => {
    expect(wrapper.contains('div.profileButton')).toBe(true);
  });

  it('expect router-link button', () => {
    expect(wrapper.contains('#Btn_id')).toBe(true);
  });

});
