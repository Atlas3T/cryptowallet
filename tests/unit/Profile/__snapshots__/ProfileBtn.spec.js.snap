// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileButton.vue renders and matches snapshot 1`] = `
<qpage-stub>
  <div
    class="profileButton"
  >
    <div
      class="container"
    >
      <router-link-stub
        event="click"
        id="Btn_id"
        tag="button"
        to="/UserProfile"
      >
        <p
          id="profile_id"
        >
          profile
        </p>
      </router-link-stub>
    </div>
  </div>
</qpage-stub>
`;
